### ISSUE TITLE
Usage metrics for dependent packages are not counted

### ISSUE THREAD

**Issue by leeederek** [kurtosis-tech/kurtosis#523]
### What's your CLI version?

0.75.3

### Description & steps to reproduce

There are 2 problems:

1. Given a local package X depends on remote package Y and Z, running local package X will only count as a single run in usage metrics, when in reality, both Y and Z remote packages were run as well. 

2. Similarly, given a local package A that depends on remote package B, running local package A will only count as a single run. We do not capture the metrics for remote package B being run at all. 

This is not great because we are losing insight/data on the usage of remote packages. We definitely care about how many times `kurtosis run` is being used, but we are getting to the point where we need the level of granularity afforded by per-package usage metrics.

### Desired behavior

Every unique run of a unique remote package is collected properly

### What is the severity of this bug?

Painful; this is causing significant friction in my workflow.

**Comment by mieubrisse** [comment]
Note: we'll definitely want to make sure that the repos that users pull from are hashed, for anonymity

**Comment by leeederek** [comment]
Yes, @mieubrisse agreed. I'm not proposing to change how that currently works and do not expect the solution here to involve changing that hashing feature/bit.
